client/shared/map/map.factory.js

```diff
@@ -1,26 +1,47 @@
 angular.module('mapModule', [])
-  .factory('mapFactory', function() {
+  .factory('mapFactory',['Track', function(Track) {
     var mapFactory = {};
     var map;
+    var userLocation;
+    var popularTracks = [];
+
     mapFactory.showMap = function() {
       map = L.map('map', {
         center: [50.6234, 26.2189],
         zoom: 14
       });
-      mapFactory.map = map;
-      return map;
-    };
 
-      /* $rootScope.map.locate({setView: true, maxZoom: 14});
+      if (userLocation) {
+        locationArea(userLocation);
+        getPopularTracks();
+      } else {
+        map.locate({setView: true, maxZoom: 14});
+      }
 
-      function onLocationFound(e) {
+      function getPopularTracks() {
+        Track.getList({location: [userLocation.lat, userLocation.lng]}).then(function(result) {
+          popularTracks = result;
+          mapFactory.popularTracks = popularTracks;
+        });
+      };
 
-          L.marker(e.latlng).addTo($rootScope.map)
-              .bindPopup("You are here").openPopup();
+      function locationArea(coords) {
+        if (userLocation) {
+          map.setView(coords, 14);
+        }
+      }
 
-          $rootScope.userLocationArea = L.circle(e.latlng, 3000).addTo($rootScope.map);
+      function onLocationFound(e) {
+        locationArea(e.latlng);
```

Comment:
use variable for e.latlng instead of calling it twice.
Each time you call e.latlng it goes to object, loop it, search appropriate key etc. if you create variable and use variable it would be chipper.